server/src/main/java/com/cloud/user/DomainManagerImpl.java

```diff
@@ -573,7 +580,8 @@ private void removeServiceOfferings(Long domainId, String domainIdString) {
         List<Long> serviceOfferingsDetailsToRemove = new ArrayList<>();
         List<ServiceOfferingJoinVO> serviceOfferingsForThisDomain = serviceOfferingJoinDao.findByDomainId(domainId);
         for (ServiceOfferingJoinVO serviceOffering : serviceOfferingsForThisDomain) {
-            if (domainIdString.equals(serviceOffering.getDomainId())) {
+            List<UserVmJoinVO> userVms = userVmJoinDao.listByServiceOfferingId(serviceOffering.getId());
```

Comment:
better to use vmInstanceDao and only get count. UserVmJoin will be more expensive and we don't need any details from it, just the count of VMs